www/data_service/src/md_demo/demo.controller.coffee

```diff
@@ -0,0 +1,10 @@
+class Demo extends Controller('app')
+    constructor: ($scope, $timeout, $log, dataService) ->
+        opened = dataService.open()
+        opened.closeOnDestroy($scope)
+        opened.getBuilders(name: 'www').then (@builders) =>
+            @builders[0].loadBuildrequests()
+
+        @unsub = ->
```

Comment:
It's just a demo :)  I wasn't thinking about it.

On Fri, Jun 19, 2015, 08:08 Chase Zhang <notifications@github.com> wrote:

> In www/data_service/src/md_demo/demo.controller.coffee
> <https://github.com/buildbot/buildbot/pull/1703#discussion_r32803500>:
>
> > @@ -0,0 +1,10 @@
> > +class Demo extends Controller('app')
> > +    constructor: ($scope, $timeout, $log, dataService) ->
> > +        opened = dataService.open()
> > +        opened.closeOnDestroy($scope)
> > +        opened.getBuilders(name: 'www').then (@builders) =>
> > +            @builders[0].loadBuildrequests()
> > +
> > +        @unsub = ->
>
> Why do you define this method in the constructor other than define it a a
> class method?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/buildbot/buildbot/pull/1703/files#r32803500>.
>